src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs

```diff
@@ -96,17 +97,28 @@ public FrameRenderer(Context context, IPropertyMapper mapper, CommandMapper comm
 			}
 		}
 
-		Size IViewHandler.GetDesiredSize(double widthMeasureSpec, double heightMeasureSpec)
+		Size IViewHandler.GetDesiredSize(double widthConstraint, double heightConstraint)
 		{
-			double minWidth = 20;
-			if (Primitives.Dimension.IsExplicitSet(widthMeasureSpec) && !double.IsInfinity(widthMeasureSpec))
-				minWidth = widthMeasureSpec;
+			var virtualView = (this as IViewHandler)?.VirtualView;
+			if (virtualView is null)
+			{
+				return Size.Zero;
+			}
+
+			var minWidth = virtualView.MinimumWidth;
+			var minHeight = virtualView.MinimumHeight;
+
+			if (!Primitives.Dimension.IsExplicitSet(minWidth))
+			{
+				minWidth = 20; // Legacy minimum value
+			}
 
-			double minHeight = 20;
-			if (Primitives.Dimension.IsExplicitSet(widthMeasureSpec) && !double.IsInfinity(heightMeasureSpec))
-				minHeight = heightMeasureSpec;
+			if (!Primitives.Dimension.IsExplicitSet(minHeight))
+			{
+				minHeight = 20; // Legacy minimum value
```

Comment:
can this be more explicit as a constant?

Reply:
Done.